python-avd/pyavd/validate_inputs.py

```diff
@@ -3,13 +3,14 @@
 # that can be found in the LICENSE file.
 from __future__ import annotations
 
+import json
 from typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
-    from .validation_result import ValidationResult
+    from .api.validation import ValidatedDataResult
 
 
-def validate_inputs(inputs: dict) -> ValidationResult:
+def validate_inputs(inputs: dict) -> ValidatedDataResult:
```

Comment:
Can we also support serialized JSON data as inputs here? Someone (us 😃) could already have the data serialized.
```suggestion
def validate_inputs(inputs: dict | str) -> ValidatedDataResult:
```

Reply:
Or maybe a file path with the data.